src/components/ConnectToQuickbooksOnlineButton/index.native.tsx

```diff
@@ -0,0 +1,80 @@
+import React, {useCallback, useState} from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import type {WebViewNavigation} from 'react-native-webview';
+import {WebView} from 'react-native-webview';
+import FullPageOfflineBlockingView from '@components/BlockingViews/FullPageOfflineBlockingView';
+import Button from '@components/Button';
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import Modal from '@components/Modal';
+import useLocalize from '@hooks/useLocalize';
+import {getQuickBooksOnlineSetupLink} from '@libs/actions/connections/QuickBooksOnline';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Session} from '@src/types/onyx';
+import type {ConnectToQuickbooksOnlineButtonProps} from './types';
+
+type ConnectToQuickbooksOnlineButtonOnyxProps = {
+    /** Session info for the currently logged in user. */
+    session: OnyxEntry<Session>;
+};
+
+const renderLoading = () => <FullScreenLoadingIndicator />;
+
+function ConnectToQuickbooksOnlineButton({policyID, session}: ConnectToQuickbooksOnlineButtonProps & ConnectToQuickbooksOnlineButtonOnyxProps) {
+    const [isWebViewOpen, setWebViewOpen] = useState(false);
+    const [isQuickbooksOnlineReady, setIsQuickbooksOnlineReady] = useState(false);
+    const {translate} = useLocalize();
+
+    const handleNavigationStateChange = useCallback(({url, loading}: WebViewNavigation) => {
+        if (loading || !url.startsWith('https://accounts.intuit.com/app/sign-in')) {
```

Comment:
Make the URL a const

Reply:
I remove this, using `startInLoadingState` with `renderLoading={renderLoading}` seems to handle this well enough